jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -273,22 +275,23 @@ ecma_prop_name_is_map_key (ecma_string_t *string_p) /**< ecma-string */
 #endif /* ENABLED (JERRY_ES2015_BUILTIN_MAP) || ENABLED (JERRY_ES2015_BUILTIN_SET) */
 
 /**
- * Allocate new ecma-string and fill it with characters from the utf8 string
+ * Checks whether a string has a special representation, that is, the string is either a magic string,
+ * an external magic string, or an uint32 number.
  *
- * @return pointer to ecma-string descriptor
+ * @return true, if string has a special representation
+ *         false, otherwise
  */
-ecma_string_t *
-ecma_new_ecma_string_from_utf8 (const lit_utf8_byte_t *string_p, /**< utf-8 string */
-                                lit_utf8_size_t string_size) /**< string size */
+static bool
+ecma_is_string_special (const lit_utf8_byte_t *string_p, /**< utf8 string */
+                        lit_utf8_size_t string_size, /**< string size */
+                        ecma_string_t **out_p) /**< [out] found special string */
```

Comment:
There should be an assert check for `out_p` imho, or a check before each `*out_p` assignment.